### PR TITLE
[Android] Implement XWalkView.removeJavascriptInterface

### PR DESCRIPTION
**Pull request by xzhan96**
Native already have the implementation, so it only need to be
exposed in XWalkView.

BUG=XWALK-6882

### CODE REVIEW

**Comment by xzhan96** [comment]
@fujunwei @lincsoon @wuhengzhi

**Comment by rakuco** [comment]
It'd be good to add one or more tests for this new API. Speaking of the API, is `XWalkAPI` really still at 6.0?

**Comment by xzhan96** [comment]
@rakuco 

> It'd be good to add one or more tests for this new API. Speaking of the API, is XWalkAPI really still at 6.0?

Thanks for reminding, test case has been added, and also correct API level to 7.0.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/AddandRemoveJavascriptInterfaceTest.java [hunk 1]
```diff
@@ -53,17 +63,25 @@ private void raisesExceptionAndSetTitle(String script) throws Throwable {
...
-    public void testAddJavascriptInterface() throws Throwable {
+    @Feature({"AddandRemoveJavascriptInterface"})
+    public void testAddandRemoveJavascriptInterface() throws Throwable {
         final String name = "add_js_interface.html";
 
         addJavascriptInterface();
         loadAssetFile(name);
         assertEquals(mExpectedStr, getTitleOnUiThread());
+
+        removeJavascriptInterface();
```
Comment (rakuco) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/AddandRemoveJavascriptInterfaceTest.java]: These tests can definitely be more comprehensive. Looking at this method, I don't know if:
- Removing an interface without reloading the page really does not have any effect.
- Removing an interface that has not been added does not cause any error.
- Adding more than 1 interface and removing one of them keeps all the others intact.

When writing tests, you really have to think of corner cases and weird possibilities, not just verify that the standard behavior works.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/AddandRemoveJavascriptInterfaceTest.java [hunk 2]
```diff
@@ -53,17 +63,25 @@ private void raisesExceptionAndSetTitle(String script) throws Throwable {
     }
 
     @SmallTest
-    @Feature({"AddJavascriptInterface"})
-    public void testAddJavascriptInterface() throws Throwable {
+    @Feature({"AddandRemoveJavascriptInterface"})
```
Comment (rakuco) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/AddandRemoveJavascriptInterfaceTest.java]: "add" and "and" are separate words, so you need to capitalize this string correctly ("AddAndRemove", not "AddandRemove").

**Comment by xzhan96** [comment]
@rakuco please help review again, thanks

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/AddAndRemoveJavascriptInterfaceTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+        addJavascriptInterface(new TestJavascriptInterface(), "testInterface");
+        loadAssetFile(name);
+        assertEquals(mExpectedStr, getTitleOnUiThread());
+
+        removeJavascriptInterface("testInterface2");
+        assertEquals(mExpectedStr, getTitleOnUiThread());
+
+        addJavascriptInterface(new TestJavascriptInterface2(), "testInterface2");
+        reloadSync(0);
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/AddAndRemoveJavascriptInterfaceTest.java]: It's bettter to use `XWalkView.RELOAD_NORMAL`, I am confused with `0`.

**Comment by wuhengzhi** [comment]
LGTM, thanks.

**Comment by xzhan96** [comment]
@rakuco please help to review, thanks

**Comment by xzhan96** [comment]
@lincsoon please help to merge, thanks

**Comment by xzhan96** [comment]
@rakuco

**Comment by wuhengzhi** [comment]
@lincsoon @xzhan96 For `addJavascriptInterface`, should it be marked as `reservable` in this PR?

**Comment by xzhan96** [comment]
I think so, @lincsoon

**Comment by axinging** [comment]
LGTM

**Comment by lincsoon** [comment]
I will add another patch to make a series of APIs reservable

**Comment by xzhan96** [comment]
> I will add another patch to make a series of APIs reservable

Ok, I update the patch again to remove the "reservable" in addJavascriptInterface.
please help to merge it later, thanks

**Comment by xzhan96** [comment]
@rakuco please help to review, thanks

**Comment by rakuco** [comment]
Sorry, I was busy/away last week. This looks much better than the version I had previously reviewed, thanks.
Does `testAddJavascriptInterfaceWithAnnotation` also need to be updated? If not, I think we are good to merge.

**Comment by xzhan96** [comment]
Thanks.

> Does  testAddJavascriptInterfaceWithAnnotation also need to be updated?

This test case is just for verify the js function without "@JavascriptInterface" should not be registered. It's unrelated with removeJavascriptInterface.

**Comment by baleboy** [comment]
Looks like we can merge
